chore: add release-please

JGAntunes opened the pull request:
This PR partially addresses #285. The idea is to provide a way to tag, release and bump the versions of both `package.json` and `swagger.yml` files. I tried running some tests on this repo but unfortunately seems like `release-please` will always create PRs against the mainline. As such I've ran some tests yesterday on a small repo - https://github.com/JGAntunes/actions-playground/pull/1 - testing `auto-commit`. The idea here is to rely on it as a workaround to bump the `swagger.yml` file based on `package.json` version (which is initially done by `release-please`).

For this to work I'll need to add a secret to the repo (currently don't have access, maybe you can help me witht it @erezrokah?), similar to what we've done, maybe we can use`netlify-bot`s access token? - https://github.com/netlify/cli/pull/1778#issue-558154582

erezrokah commented:
> For this to work I'll need to add a secret to the repo (currently don't have access, maybe you can help me witht it @erezrokah?), similar to what we've done, maybe we can use`netlify-bot`s access token? - [netlify/cli#1778 (comment)](https://github.com/netlify/cli/pull/1778#issue-558154582)

Thanks @JGAntunes I've added the secret. How are we dealing with generating the `go` client?

erezrokah commented on package.json:
```diff
@@ -17,7 +17,6 @@
   "devDependencies": {
     "analytics": "^0.2.0",
     "analytics-plugin-ga": "^0.1.5",
-    "auto-changelog": "^1.16.1",
```
I think we're missing the `package-lock.json` files changes

erezrokah commented on .github/workflows/swagger-bump.yml:
```diff
@@ -0,0 +1,20 @@
+        with:
+          node-version: 12
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm install
```
Should we use `npm ci` here?

JGAntunes replied:
I thought `npm ci` only installed `dependencies` and not `devDependencies` (which we need here) but now I'm uncertain - https://docs.npmjs.com/cli/v6/commands/npm-ci - and according to the docs doesn't seem like it. I'll change it, thanks @erezrokah 👍

erezrokah reviewed (changes requested):
Added a few comments, but will delegate to the Backend the rest of the review.

JGAntunes commented:
> Thanks @JGAntunes I've added the secret. How are we dealing with generating the go client?

I've added a more extensive comment covering that @erezrokah - https://github.com/netlify/open-api/issues/285#issuecomment-763683630

mraerino reviewed (commented):
from a process level this looks fine.

this still expects people to manually regenerate the Go client right?

since this has always been the case this is not a blocker: can we think of a process where PRs are denied when the go code is not up-to-date?
that would also block releases that had this problem (but should not if we never merged PRs with failing checks).

JGAntunes commented:
> this still expects people to manually regenerate the Go client right?

@mraerino that's correct yes, I've went through a more lengthy boring explanation here - https://github.com/netlify/open-api/issues/285#issuecomment-763683630 - but essentially we can come out with an action to verify if new files are generated through `make all` and haven't been checked in, and fail the build in that case. I would rather have it as a separate PR though. Maybe later on we can improve the automation for generating the go client, but at least that way we would already have some safeguards in place.
